src/borg/repository.py

```diff
@@ -1526,7 +1526,15 @@ def _read(self, fd, fmt, header, segment, offset, acceptable_tags, read_data=Tru
                 size, segment, offset))
         length = size - fmt.size
         if read_data:
-            data = fd.read(length)
+            try:
+                data = fd.read(length)
+            except OSError as err:
+                if err.errno == errno.EIO:
+                    raise IntegrityError(
+                        'I/O error while reading segment file [segment {}, offset {}] : {}'
+                        .format(segment, offset, fd.name)
+                    )
```

Comment:
again: i would like not to lose the information present in str(err).

Reply:
that would be `OSError: [Errno 5] Input/output error`, right? could i just use `logger.exception()` before raising?

Reply:
It would be better to have it in one log line.

Reply:
i think it would be more effective if you proposed the contents of the message.

Reply:
well, you can have a look at other places where error get catched and logged or re-raised.

but in general, using str(err) + " (whatever you want to add)" is a good approach.

hiding error information, telling something that sometimes but not always is a correct error message or suppressing errors completely is bad.

just think of "unknown error" as the extreme or a completely wrong/misleading error msg as worst examples of that.